go.mod

```diff
@@ -1,15 +1,13 @@
 module github.com/openshift/cluster-olm-operator
 
-go 1.22.5
-
-toolchain go1.22.7
+go 1.23.4
```

Comment:
openshift/api looks like this:

https://github.com/openshift/api/blob/744790f2cff777b1bb29bdccce10e4e84cff0a69/go.mod#L3-L5

Do we actually need `go 1.23.4` or can we set it to `go 1.23.0`?

Reply:
I changed this to look identical to what we have in openshift/api 👍🏽